examples/widgets/annotated_cursor.py

```diff
@@ -0,0 +1,345 @@
+        # Text positio offset
+        self.offset = np.array(offset)
+        # The axis in which the cursor position is looked up
+        self.dataaxis = dataaxis
+
+        # First call baseclass constructor.
+        # Draws cursor and remembers backround for blitting.
+        # Saves ax as class attribute.
+        super().__init__(**cursorargs)
+
+        # The numbers displayed next to cursor. Default value.
+        self.textdata = [0, 0]
+        # Default value for position of text.
+        self.setpos(self.line.get_xdata()[0], self.line.get_ydata()[0])
+        # Create invisible animated text
+        self.text = self.ax.text(
+            self.ax.get_xbound()[0],
+            self.ax.get_ybound()[0],
+            "0, 0",
+            animated=bool(self.useblit),
+            visible=False, **textprops)
+        # The position at which the cursor was last drawn
+        self.lastdrawnplotpoint = None
+
+    def onmove(self, event):
+        """Overwritten draw callback for cursor. \
+        Called when moving the mouse."""
+
+        # Leave method under the same conditions as in overwritten method
+        if self.ignore(event):
+            self.lastdrawnplotpoint = None
+            return
+        if not self.canvas.widgetlock.available(self):
+            self.lastdrawnplotpoint = None
+            return
+
+        # If the mouse left drawable area, we now make the text invisible.
+        # Baseclass will redraw complete canvas after, which makes both text
+        # and cursor disappear.
+        if event.inaxes != self.ax:
+            self.lastdrawnplotpoint = None
+            self.text.set_visible(False)
+            super().onmove(event)
+            return
+
+        # Get the coordinates, which should be displayed as text,
+        # if the event cooridnates are valid
+        plotpoint = None
+        if (event.xdata is not None) and (event.ydata is not None):
+            # Get plot point related to current x position.
+            # These coordinates are displayed in text.
+            plotpoint = self.setpos(event.xdata, event.ydata)
+            # Modify event, such that the cursor is displayed on the
+            # plotted line, not at the mouse pointer,
+            # if the returned plot point is valid
+            if plotpoint is not None:
+                event.xdata = plotpoint[0]
+                event.ydata = plotpoint[1]
+
+        # If the plotpoint is given, compare to last drawn plotpoint and
+        # return if they are the same.
+        # Skip even the call of the base class, because this would restore the
+        # background, draw the cursor lines and would leave us the job to
+        # re-draw the text.
+        if plotpoint is not None:
+            if plotpoint == self.lastdrawnplotpoint:
+                return
+
+        # Baseclass redraws canvas and cursor. Due to blitting,
+        # the added text is removed in this call, because the
+        # backround is redrawn.
```

Comment:
```suggestion
        # background is redrawn.
```

Reply:
Fixed in 0a740fc8cad97a88207777af38618adcac8bd744 in multiple spots